File: gui/python/pyvtk/cube.py
import vtk

cube = vtk.vtkCubeSource()
cube.Update()

cube_mapper = vtk.vtkPolyDataMapper()
cube_mapper.SetInputData(cube.GetOutput())

cube_actor = vtk.vtkActor()
cube_actor.SetMapper(cube_mapper)

cube_actor.GetProperty().SetColor(1.0, 0.0, 0.0)

renderer = vtk.vtkRenderer()
renderer.SetBackground(0.0, 0.0, 1.0)
renderer.AddActor(cube_actor)

render_window = vtk.vtkRenderWindow()
render_window.AddRenderer(renderer)
render_window.Render()
render_window.SetSize(400,400)
render_window.SetWindowName("OneFLOW CFD VTK Cube Test")

interactor = vtk.vtkRenderWindowInteractor()
interactor.SetRenderWindow(render_window)
interactor.Initialize()
interactor.Start()

File: gui/vtk/cube.cpp
#include <vtkCubeSource.h>
#include <vtkPolyDataMapper.h>
#include <vtkActor.h>
#include <vtkProperty.h>
#include <vtkRenderWindow.h>
#include <vtkRenderer.h>
#include <vtkRenderWindowInteractor.h>

int main(int argc, char *argv[])
{
	vtkNew<vtkCubeSource> cube;

	vtkNew<vtkPolyDataMapper> cubeMapper;
	cubeMapper->SetInputConnection( cube->GetOutputPort() );

	vtkNew<vtkActor> cubeActor;
	cubeActor->SetMapper( cubeMapper );

	cubeActor->GetProperty()->SetColor(1.0, 0.0, 0.0 );

	vtkNew<vtkRenderer> renderer;
	renderer->SetBackground( 0.0, 0.0, 1.0 );
	renderer->AddActor(cubeActor);

	vtkNew<vtkRenderWindow> renderWindow;
	renderWindow->AddRenderer(renderer);
	renderWindow->SetSize(400, 400);

	vtkNew<vtkRenderWindowInteractor> interactor;
	interactor->SetRenderWindow( renderWindow );
	interactor->Initialize();

	renderWindow->SetWindowName("OneFLOW CFD VTK Cube Test");
	renderWindow->Render();
	interactor->Start();

	return 0;
}
